src/System.IO.Pipelines/Pipe.cs

```diff
@@ -89,12 +84,11 @@ public Pipe(IBufferPool pool, PipeOptions options = null)
             }
 
             _pool = pool;
-            _readerScheduler = options.ReaderScheduler ?? InlineScheduler.Default;
-            _writerScheduler = options.WriterScheduler ?? InlineScheduler.Default;
             _maximumSizeHigh = options.MaximumSizeHigh;
             _maximumSizeLow = options.MaximumSizeLow;
-            _readerCallback = _awaitableIsNotCompleted;
-            _writerCallback = _awaitableIsCompleted;
+
+            _readerAwaitable = new Awaitable(false, options.ReaderScheduler ?? InlineScheduler.Default);
```

Comment:
Use optional parameter syntax and put the bool last. Makes it more obvious what it is.